pyquil/operator_estimation.py

```diff
@@ -799,17 +850,169 @@ def measure_observables(qc: QuantumComputer, tomo_experiment: TomographyExperime
+                    total_counts=n_shots,
+                    calibration_expectation=obs_calibr_mean.item(),
+                    calibration_stddev=np.sqrt(obs_calibr_var).item(),
+                    calibration_counts=calibr_shots,
+                    corrected_expectation=corrected_mean.item(),
+                    corrected_stddev=np.sqrt(corrected_var).item(),
+                )
+
+            else:
+                # No calibration
+                obs_calibr_mean = None
+                obs_calibr_var = None
+                calibr_shots = None
+                corrected_mean = None
+                corrected_var = None
+
+                yield ExperimentResult(
+                    setting=setting,
+                    expectation=obs_mean.item(),
+                    stddev=np.sqrt(obs_var).item(),
+                    total_counts=n_shots,
+                )
+
+
+def _ops_strs_symmetrize(qubits: List[int]) -> List[str]:
+    """
+    :param qubits: list specifying the qubits whose readout errors we wish to symmetrize
+    :return: list with the operation strings necessary for exhaustive symmetrization
+    """
+    ops_strings = []
+    for prod in itertools.product(['I', 'X'], repeat=len(qubits)):
+        ops_strings.append(''.join(prod))
+    return ops_strings
+
+
+def _ops_str_to_prog(ops_str: str, qubits: List[int]) -> Program:
+    """
+    :param ops_str: string specifying the operation to be carried out on `qubits`
+    :param qubits: list specifying the qubits to be carried operations on
+    :return: Program with the operations specified in `ops_str` on the qubits specified in `qubits`
+    """
+    assert len(ops_str) == len(qubits), "Mismatch of qubits and operations"
+    prog = Program()
+    for i, op_ch in enumerate(ops_str):
+        if op_ch == 'I':
+            continue
+        elif op_ch == 'X':
+            prog += Program(X(qubits[i]))
+        else:
+            raise ValueError("ops_strings_ should only consist of 'I's and/or 'X's")
+    return prog
+
+
+def _ops_str_to_flips(ops_str: str, qubits: List[int]) -> Dict:
+    """
+    :param ops_str: string specifying the operation to be carried out on `qubits`
+    :param qubits: list specifying the qubits to be carried operations on
+    :return: Dict specyfing whether to flip the readout results or not, depending on
+        the operations specified in `ops_str`, which in turn are operating on the
+        qubits specified in `qubits`
+    """
+    d_flip = {}
+    for i, op_ch in enumerate(ops_str):
+        q = qubits[i]
+        if op_ch == 'I':
+            d_flip[q] = 0
+        elif op_ch == 'X':
+            d_flip[q] = 1
+        else:
+            raise ValueError("ops_strings_ should only consist of 'I's and/or 'X's")
+    return d_flip
+
+
+def _stack_dicts(dict1: Dict, dict2: Dict) -> Dict:
+    """
+    :param dict1: Dict keyed with integer specifying qubit, valued by 1-dimensional numpy array specifying
```

Comment:
is this wrapped to 100 characters? it's overflowing when I view the split diff

Reply:
My screen was telling me it was fine, but I've wrapped to shorter width